tests/components/google_travel_time/test_sensor.py

```diff
@@ -0,0 +1,242 @@
+"""Test the Google Maps Travel Time sensors."""
+
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.google_travel_time.const import (
+    CONF_ARRIVAL_TIME,
+    CONF_DEPARTURE_TIME,
+    CONF_TRAVEL_MODE,
+    DOMAIN,
+)
+
+from .const import MOCK_CONFIG
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(name="mock_update")
+def mock_update_fixture():
+    """Mock an update to the sensor."""
+    with patch("homeassistant.components.google_travel_time.sensor.Client"), patch(
+        "homeassistant.components.google_travel_time.sensor.distance_matrix",
+        return_value={
+            "rows": [
+                {
+                    "elements": [
+                        {
+                            "duration_in_traffic": {
+                                "value": 1620,
+                                "text": "27 mins",
+                            },
+                            "duration": {
+                                "value": 1560,
+                                "text": "26 mins",
+                            },
+                            "distance": {"text": "21.3 km"},
+                        }
+                    ]
+                }
+            ]
+        },
+    ):
+        yield
+
+
+@pytest.fixture(name="mock_update_duration")
+def mock_update_duration_fixture():
+    """Mock an update to the sensor returning no duration_in_traffic."""
+    with patch("homeassistant.components.google_travel_time.sensor.Client"), patch(
+        "homeassistant.components.google_travel_time.sensor.distance_matrix",
+        return_value={
+            "rows": [
+                {
+                    "elements": [
+                        {
+                            "duration": {
+                                "value": 1560,
+                                "text": "26 mins",
+                            },
+                            "distance": {"text": "21.3 km"},
+                        }
+                    ]
+                }
+            ]
+        },
+    ):
+        yield
+
+
+@pytest.fixture(name="mock_update_empty")
+def mock_update_empty_fixture():
+    """Mock an update to the sensor with an empty response."""
+    with patch("homeassistant.components.google_travel_time.sensor.Client"), patch(
+        "homeassistant.components.google_travel_time.sensor.distance_matrix",
+        return_value=None,
+    ):
+        yield
+
+
+@pytest.mark.usefixtures("mock_update")
+async def test_sensor(hass):
```

Comment:
ha, I've never seen fixtures being used this way. I always use it like this:

```suggestion
async def test_sensor(hass, mock_update):
```

Reply:
The annotation eases my pain of seing an unused variable 😅
I've seen several other integrations use this style.

So I understood I shouldn't use the annotation style?

Reply:
I'm fine either way. We don't run pylint on tests.